From 33283e88362436a9bb30d39a2615609674f4aac6 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Mon, 14 Jul 2014 10:38:18 -0400
Subject: [fs] ext4: fix type declaration of ext4_validate_block_bitmap

Message-id: <1405334298-26039-2-git-send-email-lczerner@redhat.com>
Patchwork-id: 85732
O-Subject: [RHEL 7.1 PATCH 2/2] ext4: fix type declaration of ext4_validate_block_bitmap
Bugzilla: 1091055
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: "Darrick J. Wong" <darrick.wong@oracle.com>

BZ 1091055
https://bugzilla.redhat.com/show_bug.cgi?id=1091055

BREW 7657226
https://brewweb.devel.redhat.com/taskinfo?taskID=7657226

Upstream commit dbde0abed8c6e9e938c2194675ce63f5769b0d37

 The block_group parameter to ext4_validate_block_bitmap is both used
 as a ext4_group_t inside the function and the same type is passed in
 by all callers.  We might as well use the typedef consistently instead
 of open-coding the 'unsigned int'.

 Signed-off-by: Darrick J. Wong <darrick.wong@oracle.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/balloc.c b/fs/ext4/balloc.c
index b430afe..a40b1f9 100644
--- a/fs/ext4/balloc.c
+++ b/fs/ext4/balloc.c
@@ -305,7 +305,7 @@ struct ext4_group_desc * ext4_get_group_desc(struct super_block *sb,
  */
 static ext4_fsblk_t ext4_valid_block_bitmap(struct super_block *sb,
          struct ext4_group_desc *desc,
-         unsigned int block_group,
+         ext4_group_t block_group,
          struct buffer_head *bh)
 {
  ext4_grpblk_t offset;
@@ -352,7 +352,7 @@ static ext4_fsblk_t ext4_valid_block_bitmap(struct super_block *sb,
 
 void ext4_validate_block_bitmap(struct super_block *sb,
           struct ext4_group_desc *desc,
-          unsigned int block_group,
+          ext4_group_t block_group,
           struct buffer_head *bh)
 {
  ext4_fsblk_t blk;
diff --git a/fs/ext4/ext4.h b/fs/ext4/ext4.h
index 8586e8e..1f33901 100644
--- a/fs/ext4/ext4.h
+++ b/fs/ext4/ext4.h
@@ -1921,7 +1921,7 @@ extern ext4_group_t ext4_get_group_number(struct super_block *sb,
 
 extern void ext4_validate_block_bitmap(struct super_block *sb,
            struct ext4_group_desc *desc,
-           unsigned int block_group,
+           ext4_group_t block_group,
            struct buffer_head *bh);
 extern unsigned int ext4_block_group(struct super_block *sb,
    ext4_fsblk_t blocknr);
-- 
1.7.1